osquery/tables/system/darwin/xattr.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <vector>
+#include <string>
+
+#include <sys/xattr.h>
+
+#include <glog/logging.h>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+
+namespace osquery {
+namespace tables {
+
+std::string handleError(){
+  switch(errno){
+    case ENOATTR:
+      return "No such attribute";
+    case ENOTSUP:
+      return "No system support, or support disabled";
+    case ERANGE:
+      return "Size too small to hold attribute";
+    case EPERM:
+      return "The named attribute is not permitted for this type of object";
+    case EINVAL:
+      return "Name is invalid, or options has an unsupported bit set";
+    case EISDIR:
+      return "The path is not a regular file";
+    case ENOTDIR:
+      return "Part of the path was not a directory";
+    case ENAMETOOLONG:
+      return "The filename was too long";
+    case EACCES:
+      return "Insufficient permissions to read the requested file";
+    case ELOOP:
+      return "Too many symbolic links were encountered";
+    case EFAULT:
+      return "The path or name is invalid";
+    case EIO: 
+      return "There was an I/O error while reading";
+    default:
+      return "No data";
+  }
+}
+
+//Broken out into it's own function so we can use it to get attributes later
+char* getAttribute(const boost::filesystem::path path, const char* attribute, int &ret, int &buf_len){
+    buf_len = getxattr(path.string().c_str(), attribute, NULL, 0, 0 , 0);
+    char *buffer = (char*)malloc(buf_len);
+    ret = getxattr(path.string().c_str(), attribute, buffer, buf_len, 0 , 0);
+    return buffer;
+}
+
+QueryData genXattr(QueryContext& context) {
+  QueryData results;
+  auto paths = context.constraints["path"].getAll(EQUALS);
+
+  for (const auto& path_string : paths) {
+    boost::filesystem::path path = path_string;
+    if (!boost::filesystem::is_regular_file(path)) {
```

Comment:
You don't want to handle directories? You could support recursive searching if someone asked for /Users/<username>/Downloads.